replication-api-impl/src/main/java/com/connexta/replication/api/impl/worker/Worker.java

```diff
@@ -0,0 +1,354 @@
+            "Received task to create metadata on site {}, but it was deleted on {}",
+            destination.getSystemName(),
+            source.getSystemName());
+        task.fail(
+            ErrorCode.NO_LONGER_EXISTS,
+            String.format(
+                "Metadata deleted from site %s before able to create", source.getSystemName()));
+        return;
+      }
+    } catch (AdapterInterruptedException e) {
+      throw new InterruptedException(e.getMessage());
+    } catch (AdapterException e) {
+      LOGGER.debug("Failed to check if resource exists", e);
+      task.fail(
+          ErrorCode.OPERATION_FAILURE,
+          String.format("Failed check resource existence on %s", destination.getSystemName()));
+      return;
+    }
+
+    URI uri = task.getResource().flatMap(ResourceInfo::getResourceUri).orElse(null);
+    if (uri != null) {
+      ResourceRequest request = new ResourceRequestImpl(metadata);
+      ResourceResponse response;
+      try {
+        response = source.readResource(request);
+      } catch (AdapterInterruptedException e) {
+        throw new InterruptedException(e.getMessage());
+      } catch (AdapterException e) {
+        LOGGER.debug("Failed to read resource", e);
+        task.fail(
+            ErrorCode.OPERATION_FAILURE,
+            String.format("Failed to read resource from %s", source.getSystemName()));
+        return;
+      }
+
+      try {
+        Resource resource = response.getResource();
+        boolean success = destination.createResource(new CreateStorageRequestImpl(resource));
+        if (!success) {
+          task.fail(
+              ErrorCode.OPERATION_FAILURE,
+              String.format(
+                  "Failed to %s resource %s from %s to %s",
+                  operation, uri, source.getSystemName(), destination.getSystemName()));
+          return;
+        }
+      } catch (AdapterInterruptedException e) {
+        throw new InterruptedException(e.getMessage());
+      } catch (AdapterException e) {
+        LOGGER.debug("Failed to create resource", e);
+        task.fail(
+            ErrorCode.OPERATION_FAILURE,
+            String.format(
+                "Failed to create resource from %s to %s",
+                source.getSystemName(), destination.getSystemName()));
+        return;
+      }
+    } else {
+      try {
+        LOGGER.info("No resource URI specified for task {}. Doing metadata create only", task);
+        CreateRequest request = new CreateRequestImpl(List.of(metadata));
+        boolean success = destination.createRequest(request);
+        if (!success) {
+          task.fail(
+              ErrorCode.OPERATION_FAILURE,
+              String.format(
+                  "Failed to %s metadata %s from %s to %s",
+                  operation, metadataId, source.getSystemName(), destination.getSystemName()));
+          return;
+        }
+      } catch (AdapterInterruptedException e) {
+        throw new InterruptedException(e.getMessage());
```

Comment:
✏️ Don't bother with the message since it won't ever have one. Either create it blank or pass the exception as a cause only.